pkg/pipeline/BuildPipelineConfigService.go

```diff
@@ -1792,10 +1793,17 @@ func (impl *PipelineBuilderImpl) RetrieveArtifactsByCDPipeline(pipeline *pipelin
 		parentCdId = parentId
 	}
 
-	if stage == bean2.CD_WORKFLOW_TYPE_DEPLOY && len(pipeline.PreStageConfig) > 0 {
-		// Parent type will be PRE for DEPLOY stage
-		parentId = pipeline.Id
-		parentType = bean2.CD_WORKFLOW_TYPE_PRE
+	if stage == bean2.CD_WORKFLOW_TYPE_DEPLOY {
+		pipelinePreStage, err := impl.pipelineStageRepository.GetCdStageByCdPipelineIdAndStageType(pipeline.Id, repository3.PIPELINE_STAGE_TYPE_PRE_CD)
+		if err != nil && err != pg.ErrNoRows {
+			impl.logger.Errorw("error in fetching PRE-CD stage by cd pipeline id", "pipelineId", pipeline.Id, "err", err)
+			return nil, err
+		}
+		if pipelinePreStage != nil || len(pipeline.PreStageConfig) > 0 {
```

Comment:
GetCdStageByCdPipelineIdAndStageType will always give  pipelinePreStage as not nill. Need to add check for pipelinePreStage.Id > 0 to be completely sure.